packages/encoding/src/rfc3339.ts

```diff
@@ -40,7 +40,12 @@ export function fromRfc3339(str: string): Date {
 
   const tzOffset = tzOffsetSign * (tzOffsetHours * 60 + tzOffsetMinutes) * 60; // seconds
 
-  const timestamp = Date.UTC(year, month - 1, day, hour, minute, second, milliSeconds) - tzOffset * 1000;
+  let timestamp = Date.UTC(year, month - 1, day, hour, minute, second, milliSeconds);
+
+  // Date.UTC maps year 0-99 to 1900-1999. Ensure the correct year is set and THEN apply the offset
+  const date = new Date(timestamp);
+  timestamp = date.setUTCFullYear(year) - tzOffset * 1000;
+
```

Comment:
Alright, let's see. Few observations here:

- A time before epoch seems well supported:
  > The maximum timestamp representable by a Date object is slightly smaller than the maximum safe integer ([Number.MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER), which is 9,007,199,254,740,991). A Date object can represent a maximum of ±8,640,000,000,000,000 milliseconds, or ±100,000,000 (one hundred million) days, relative to the epoch. This is the range from April 20, 271821 BC to September 13, 275760 AD. Any attempt to represent a time outside this range results in the Date object holding a timestamp value of [NaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN), which is an "Invalid Date".
- The two year issue seems to be a known problem: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date?retiredLocale=de#interpretation_of_two-digit_years and
  > Note: Some methods, including the Date() constructor, Date.UTC(), and the deprecated [getYear()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getYear)/[setYear()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setYear) methods, interpret a two-digit year as a year in the 1900s. For example, new Date(99, 5, 24) is interpreted as June 24, 1999, not June 24, 99. See [Interpretation of two-digit years](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date?retiredLocale=de#interpretation_of_two-digit_years) for more information.

So this all makes sense. Now I think the code it a bit hard to follow with using `Date.UTC` first and then patching its broken behaviour. Can we instead create an empty Date (e.g. `new Date()`) and then call `setUTCFullYear(yearValue, monthValue, dateValue)`, `setUTCHours(hoursValue, minutesValue, secondsValue, msValue)`. Then apply the offset. I think the current code is correct, but confusing to audit.

Reply:
Why the explicit implementation btw instead of simply using `Date.parse()` ?

Reply:
Good question. I think the main reason was nanosecond support in the fractional part. But it might also be because of implementation inconsistencies.